resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -647,6 +656,7 @@ private[spark] class Client(
         val fileBase64 = Base64.encodeBase64String(fileBytes)
         UploadedAppResource(resourceBase64Contents = fileBase64, name = appFile.getName)
       case "local" => ContainerAppResource(mainAppResource)
+      case "nop" => NopAppResource()
```

Comment:
this nop resource I think is the same concept as Spark already has in `SparkLauncher#NO_RESOURCE`

Seems like this bit of your PR could be slightly modified to address  https://github.com/apache-spark-on-k8s/spark/issues/213 ?